sample/dump.c

```diff
@@ -112,6 +170,12 @@ int main() {
         rmt_LogText("start profiling");
         delay();
         rmt_LogText("end profiling");
+
+        rmt_PropertySet_Bool(WasUpdated, RMT_TRUE);
+        rmt_PropertyAdd_U32(FrameCounter, 1);
+
+        rmt_PropertySnapshotAll();
```

Comment:
@dwilliamson : ⚠️ I haven't figured out the actual issue yet, but when adding this call, the remotery destructor fails  on `Delete(ObjectAllocator, rmt->propertyAllocator);`:
 ```
 Assertion failed: (allocator->nb_inuse == 0), function ObjectAllocator_Destructor, file lib/Remotery.c, line 2132.
 ```
 
The count is 6, which I assume might be the root property + my 5 properties in dump.c.

Tested with:
```
$ clang -std=c89 -DRMT_ENABLED=1 -DRMT_ARCH_64BIT -g -O0 -Ilib sample/dump.c lib/Remotery.c -o test && ./test
```

Reply:
I believe it's because the snapshot data is still in flight, and we're shutting down the Remotery instance before that call is handled.

Reply:
Hmmm... `BIN_ERROR_CHECK` will leak memory in `Remotery_SendPropertySnapshot` if if fails any write, not calling `FreePropertySnapshots`. Will have to pull the serialisation into a separate function.

Reply:
Found the shutdown leak, as well...